dashboard/src/app/workspaces/workspace-details/environments/list-ports/edit-port-dialog/edit-port-dialog.controller.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+/**
+ * @ngdoc controller
+ * @name list.environment.variables.controller:EditPortDialogController
+ * @description This class is handling the controller for the dialog box about editing the environment variable.
+ * @author Oleksii Kurinnyi
+ */
+export class EditPortDialogController {
+
+  /**
+   * Default constructor that is using resource
+   * @ngInject for Dependency injection
+   */
+  constructor($mdDialog, $scope, lodash) {
+    this.$mdDialog = $mdDialog;
+    this.lodash = lodash;
+
+    this.updateInProgress = false;
+    this.port = parseInt(this.servers[this.serverName].port,10);
+    this.protocol = this.servers[this.serverName].protocol;
+
+    this.existServers = angular.copy(this.servers);
+    delete this.existServers[this.serverName];
+
+    let ctrl = this;
+    // validate port uniqueness
+    $scope.isUnique = (port) => {
+      let isUsed = ctrl.lodash.some(ctrl.existServers, (server) => {
+        return parseInt(server.port, 10) === port;
+      });
+      return !isUsed;
```

Comment:
maybe we don't need to introduce local var isUsed (up to you)
but would be better to not use the $scope, and use controller